### PR TITLE
Add FIM template for Codestral-22B

### PR DESCRIPTION
**Pull request by AndrewRocky**
NOT TESTED! I don't have anything set up for creating VSCode extensions (not even TS... compiler?)

### tl;dr: 
I've tried adding Codestral into FIM templates.

BTW, it seems like it's FIM tuning is meh? Or maybe it's a quants issue(5bpw)? Anyway, back to starcoder2-15B.

### Description
This is implementation of Codestral's FIM template.

Template itself is slightly different compared to Continue's implementaion (they use something like `[SUFFIX]{suffix}[PREFIX]{filecontext}{prefix}`) - I've put `fileContext` before `[SUFFIX]` - because as prefix grows Codestral quickly starts to lose suffix's context and use filecontext instead

Also it seems like Codestral prefers `+++++ relative/file.path` as a file header? I don't think I've seen anything about it in official docs though.

I've tested this template directly in oobabooga on `bartowski_Codestral-22B-v0.1-exl2_5_0` (5 bpw model - enough for 24gb with 32k context) (but I've used  `+++++ relative/file.path` as file headers). Tests were very limited, from my findings Codestral:
* heavily respects PREFIX
* respects SUFFIX in easy cases (if PREFIX even slightly contradicts SUFFIX - result won't adhere to SUFFIX)
* often ignores fileContext

Anyway, I haven't tested this code. And I don't mind if it won't be merged - I didn't like Codestral FIM performance anyway :P

### CODE REVIEW

**Comment by rjmacarthy** [comment]
LGTM, we can merge and see if anyone reports it's broken, I don't use codestral for FIM either...
